compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/JvmBackendContext.kt

```diff
@@ -87,6 +87,10 @@ class JvmBackendContext(
 
     internal val localDelegatedProperties = mutableMapOf<IrClass, List<IrLocalDelegatedPropertySymbol>>()
 
+    // If the JVM fqname of a class differs from what is implied by its parent, e.g. if it's a file class
+    // annotated with @JvmPackageName, the correct name is recorded here.
+    internal lateinit var classNameOverride: MutableMap<IrClass, FqName>
```

Comment:
I think it'd be a bit more convenient to store `JvmClassName` (`/`-separated) as values. Then the call site at `FileClassLowering` can use this:
```
context.classNameOverride[this] = JvmClassName.byInternalName(partClassType.internalName)
```
and the call site at `IrTypeMapper` can use this:
```
context.classNameOverride[irClass]?.let { return it.internalName }
```
eliminating the manual `.`-`/` conversions

Reply:
Done.